src/main/resources/update/gitbucket-core_4.32.xml

```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<changeSet>
+  <!-- add ID column to REPOSITORY table -->
+
+  <!-- drop other table's foreign keys -->
+  <dropForeignKeyConstraint constraintName="IDX_ACTIVITY_FK0" baseTableName="ACTIVITY"/>
+  <dropForeignKeyConstraint constraintName="IDX_COLLABORATOR_FK0" baseTableName="COLLABORATOR"/>
+  <dropForeignKeyConstraint constraintName="IDX_COMMIT_COMMENT_FK0" baseTableName="COMMIT_COMMENT"/>
+  <dropForeignKeyConstraint constraintName="IDX_COMMIT_STATUS_FK1" baseTableName="COMMIT_STATUS"/>
+  <dropForeignKeyConstraint constraintName="IDX_LABEL_FK0" baseTableName="LABEL"/>
+  <dropForeignKeyConstraint constraintName="IDX_MILESTONE_FK0" baseTableName="MILESTONE"/>
+  <dropForeignKeyConstraint constraintName="IDX_ISSUE_FK0" baseTableName="ISSUE"/>
+  <dropForeignKeyConstraint constraintName="IDX_ISSUE_ID_FK1" baseTableName="ISSUE_ID"/>
+  <dropForeignKeyConstraint constraintName="IDX_WEB_HOOK_FK0" baseTableName="WEB_HOOK"/>
+  <dropForeignKeyConstraint constraintName="IDX_PROTECTED_BRANCH_FK0" baseTableName="PROTECTED_BRANCH"/>
+  <dropForeignKeyConstraint constraintName="IDX_DEPLOY_KEY_FK0" baseTableName="DEPLOY_KEY"/>
+  <dropForeignKeyConstraint constraintName="IDX_PRIORITY_FK0" baseTableName="PRIORITY"/>
+  <dropForeignKeyConstraint constraintName="IDX_RELEASE_TAG_FK0" baseTableName="RELEASE_TAG"/>
+
+  <!-- modify REPOSITORY table -->
+  <dropForeignKeyConstraint constraintName="IDX_REPOSITORY_FK0" baseTableName="REPOSITORY"/>
+  <dropPrimaryKey tableName="REPOSITORY" constraintName="IDX_REPOSITORY_PK"/>
+  <addColumn tableName="REPOSITORY">
+    <column name="REPOSITORY_ID" type="int" nullable="false" autoIncrement="false" unique="true"/>
```

Comment:
Probably, adding non-nullable column without default value will cause a migration error. In this case, we have to add nullable column, fill column values then we can add a non-null constraint.